src/libasr/pass/replace_symbolic.cpp

```diff
@@ -559,6 +559,84 @@ class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisi
                 process_unary_operator(al, loc, x, module_scope, "basic_expand", target);
                 break;
             }
+            case LCompilers::ASRUtils::IntrinsicScalarFunctions::SymbolicGetArgument: {
+                // Define necessary function symbols
+                ASR::expr_t* value1 = handle_argument(al, loc, x->m_args[0]);
+                ASR::symbol_t* basic_get_args_sym = declare_basic_get_args_function(al, loc, module_scope);
+                ASR::symbol_t* vecbasic_new_sym = declare_vecbasic_new_function(al, loc, module_scope);
+                ASR::symbol_t* vecbasic_get_sym = declare_vecbasic_get_function(al, loc, module_scope);
+                ASR::symbol_t* vecbasic_size_sym = declare_vecbasic_size_function(al, loc, module_scope);
+
+                // Define necessary variables
+                ASR::ttype_t* CPtr_type = ASRUtils::TYPE(ASR::make_CPtr_t(al, loc));
+                ASR::symbol_t* args_sym = ASR::down_cast<ASR::symbol_t>(ASR::make_Variable_t(
+                    al, loc, current_scope, s2c(al, "args"), nullptr, 0, ASR::intentType::Local,
+                    nullptr, nullptr, ASR::storage_typeType::Default, CPtr_type, nullptr,
+                    ASR::abiType::BindC, ASR::Public, ASR::presenceType::Required, false));
+                //if (!current_scope->get_symbol("args")) {
+                current_scope->add_symbol(s2c(al, "args"), args_sym);
```

Comment:
I see 2 things to improve in this pr. This is one of them. What we want to do here is 
```
    args: CPtr = vecbasic_new()
    basic_get_args(x, args)
```
Hence I've created a local `args_sym` a symbol using the name "args". But I am not sure about this as to should we use a unique name of some sort here, args might be common which a user might end up using.

Secondly I think I am not able to use the `args` attribute for the second time due to some reason related to how I add it to the `current_scope`. Hence if we extract twice it gives an error 
```
    arg1: S = x.args[1]
    arg2: S = x.args[1]
    print(arg1)
    print(arg2)
```
```
(lf) anutosh491@spbhat68:~/lpython/lpython$ lpython --enable-symengine examples/expr2.py 
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  Binary file "/home/anutosh491/lpython/lpython/src/bin/lpython", in _start()
  Binary file "/lib/x86_64-linux-gnu/libc.so.6", in __libc_start_main()
  File "/home/anutosh491/lpython/lpython/src/bin/lpython.cpp", line 1869, in ??
...............
LCompilers::ASR::IntrinsicScalarFunction_t*, LCompilers::SymbolTable*, LCompilers::ASR::expr_t*)
    current_scope->add_symbol(s2c(al, "args"), args_sym);
AssertFailed: scope.find(name) == scope.end()
```
Using it once works. @Thirumalai-Shaktivel could you maybe help me out here!

Reply:
I think we can't reuse the args local variable that we would like to extract some argument multiple times (for the same expression or different expressions) hence we would need a unique variable here.
```
(lf) anutosh491@spbhat68:~/lpython/lpython$ cat examples/expr2.py 
from lpython import S
from sympy import Symbol

def main0():
    x: S = Symbol("x")
    y: S = Symbol("y")
    x = x**y
    y = x + y
    arg1: S = x.args[0]
    arg2: S = x.args[1]
    arg3: S = y.args[0]
    print(arg1)
    print(arg2)
    print(arg3)

if __name__ == "__main__":
   main0()
(lf) anutosh491@spbhat68:~/lpython/lpython$ lpython --enable-symengine examples/expr2.py 
x
y
y